ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/events/EventsTableInsert.scala

```diff
@@ -111,23 +111,24 @@ private[events] trait EventsTableInsert { this: EventsTable =>
   }
 
   private case class AccumulatingBatches(
-      creates: Vector[RawBatch.Event.Created],
-      exercises: Vector[RawBatch.Event.Exercised],
+      creates: Vector[RawBatch.Event[RawBatch.Event.Created]],
+      exercises: Vector[RawBatch.Event[RawBatch.Event.Exercised]],
       archives: Vector[Vector[NamedParameter]],
   ) {
 
-    def add(create: RawBatch.Event.Created): AccumulatingBatches =
+    def add(create: RawBatch.Event[RawBatch.Event.Created]): AccumulatingBatches =
       copy(creates = creates :+ create)
 
-    def add(exercise: RawBatch.Event.Exercised): AccumulatingBatches =
+    def add(exercise: RawBatch.Event[RawBatch.Event.Exercised])(
+        implicit dummy: DummyImplicit): AccumulatingBatches =
```

Comment:
What is this dummy implicit for?

Reply:
It makes the first two `add` overloads have different erasures in bytecode.

Reply:
Mh, ok. Would a simple change of name have worked, though?

Reply:
Yes, it would.

We do something similar for the superclass of all Scala-codegenned record types, since we have methods share names with possible field names:

https://github.com/digital-asset/daml/blob/e23a488596d412b804dc4fd7af821d2c5b8dd5df/language-support/scala/bindings/src/main/scala/com/digitalasset/ledger/client/binding/binding/Template.scala#L22-L25